ui/ui-frontend/projects/archive-search/src/app/archive/archive-search/archive-search.component.ts

```diff
@@ -946,101 +973,115 @@ export class ArchiveSearchComponent implements OnInit, OnChanges, OnDestroy {
   selectedItemCountKnown(): boolean {
     return !this.waitingToGetFixedCount || !this.isAllchecked;
   }
-  prepareToLaunchVitamAction(actionType: string) {
+
+  async prepareToLaunchVitamAction() {
     if (!this.selectedItemCountKnown()) {
       if (this.criteriaSearchList && this.criteriaSearchList.length > 0) {
         this.pendingGetFixedCount = true;
         this.submitedGetFixedCount = true;
-        this.archiveService
-          .getTotalTrackHitsByCriteria(this.criteriaSearchList, this.accessContract)
-          .subscribe((exactCountResults: number) => {
-            if (exactCountResults !== -1) {
-              this.totalResults = exactCountResults;
-              if (this.isAllchecked) {
-                this.itemSelected = this.totalResults - this.itemNotSelected;
-              }
-              this.waitingToGetFixedCount = false;
-              this.managementRulesSharedDataService.emitHasExactCount(true);
-            }
-            this.pendingGetFixedCount = false;
-
-            this.launchVitamAction(actionType);
-          });
+        const exactCountResults: number = await this.archiveService.getTotalTrackHitsByCriteria(this.criteriaSearchList, this.accessContract)
+          .toPromise();
```

Comment:
pourquoi to promise ?

Reply:
ce n'est pas ce qu'il y a de plus beau mais dans ce petit cas ca me permet d'await ce petit appel ^^